core/imageroot/var/lib/nethserver/cluster/actions/set-smarthost/01validate_settings

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+#
+# Copyright (C) 2022 Nethesis S.r.l.
+# http://www.nethesis.it - nethserver@nethesis.it
+#
+# This script is part of NethServer.
+#
+# NethServer is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License,
+# or any later version.
+#
+# NethServer is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NethServer.  If not, see COPYING.
+#
+
+import agent
+import agent.tasks
+import os
+import sys
+import json
+import smtplib
+import ssl
+
+
+data = json.load(sys.stdin)
+
+port =  int(data['port'])
+host = data['host']
+username = data['username']
+password = data['password']
+tls = data['tls']
+tls_verify = data['tls_verify']
+
+# verify the hostname and the certificate
+ctx = ssl.create_default_context()
+if tls_verify:
+    ctx.check_hostname = True
+    ctx.verify_mode = ssl.CERT_REQUIRED
+else:
+    ctx.check_hostname = False
+    ctx.verify_mode = ssl.CERT_NONE
+
+try :
+    smtp = smtplib.SMTP(host, port=port, timeout=10)
+    # Enable starttls
+    if tls:
+        smtp.starttls(context=ctx)
+    smtp.login(username, password)
```

Comment:
Make `login` conditional.

Only if the `username` value is returned by the SmartHost class the SMTP authentication (LOGIN) must be enforced.

This rule should be documented too.